### DISCUSSION TITLE
How is OAuth supposed to work?

### DISCUSSION THREAD

**Discussion by astrolope** [grafana/grafana-infinity-datasource#310] [Q&A]
We have configured client ID and secret OAuthentication into our datasource. 

Is there supposed to be a redirect prompting login to happen on query? 
If so, what should the redirect URL be? With it currently setup all requests are failing. 

Some direction here would be greatly appreciated.

**Comment by yesoreyeram** [comment]
There are different type of oauth2 workflows exist. 'client_credentials' is one of them. ( This doesn't require any human interaction like redirection/approval). Infinity currently only supports it.

**Reply by astrolope** [reply]
Where does the token info endpoint come into play then? I tried to read through the code but seeing the httpClient getting re-defined with the GO oauth library I don't see where the actual response from our oauth server gets appended into this client.

**Reply by yesoreyeram** [reply]
See if this helps https://oauth.net/2/grant-types/client-credentials/
And https://pkg.go.dev/golang.org/x/oauth2/clientcredentials
